package/MDAnalysis/core/groups.py

```diff
@@ -3106,7 +3098,7 @@ def write(self, filename=None, file_format="PDB",
         try:
             # here `file_format` is only used as default,
             # anything pulled off `filename` will be used preferentially
-            writer = get_selection_writer_for(filename, file_format)
+            writer = get_selection_writer_for(filename, file_format or 'PDB')
```

Comment:
I don't like `file_format or 'PDB'`. Better be explicit so that only `None` really gets defaulted:
```python
file_format if file_format is not None else 'PDB'
```